controllers/common.go

```diff
@@ -19,50 +19,31 @@ import (
 
 var hashOptions = hashstructure.HashOptions{ZeroNil: true}
 
-// isSameApiDefinition calculates the hash of two ApiDefinitionSpec structs and returns true
-// if hashes are equal. It is used to prevent calling extra Update requests to Tyk Gateway
-// if there is no changes on resources.
-func isSameApiDefinition(apiDef1, apiDef2 *model.APIDefinitionSpec) bool {
-	api1Hash, err1 := hashstructure.Hash(apiDef1, hashstructure.FormatV2, &hashOptions)
+// calculateHashes calculates hashes of given two interfaces. Returns empty string for hash
+// if any error occurs during calculation.
+func calculateHashes(i1, i2 interface{}) (hash1, hash2 string) {
+	h1, err1 := hashstructure.Hash(i1, hashstructure.FormatV2, &hashOptions)
 	if err1 == nil {
-		api2Hash, err2 := hashstructure.Hash(apiDef2, hashstructure.FormatV2, &hashOptions)
-		if err2 == nil {
-			return api1Hash == api2Hash
-		}
+		hash1 = strconv.FormatUint(h1, 10)
 	}
 
-	return false
-}
-
-const (
-	proTagName = "pro"
-	ossTagName = "oss"
-)
-
-func hasherTagName(ctx context.Context) string {
-	r := tykClient.GetContext(ctx)
-	if r.Env.Mode == "pro" {
-		return proTagName
+	h2, err2 := hashstructure.Hash(i2, hashstructure.FormatV2, &hashOptions)
+	if err2 == nil {
+		return hash1, strconv.FormatUint(h2, 10)
 	}
 
-	return ossTagName
+	return
 }
 
-// isSameSecurityPolicy calculates the hash of two SecurityPolicySpec structs and returns true
-// if hashes are equal. It is used to prevent calling extra Update requests to Tyk Gateway
-// if there is no changes on resources.
-func isSameSecurityPolicy(ctx context.Context, policy1, policy2 *v1alpha1.SecurityPolicySpec) bool {
-	myHashOptions := hashstructure.HashOptions{ZeroNil: true, TagName: hasherTagName(ctx)}
-
-	pol1Hash, err1 := hashstructure.Hash(policy1, hashstructure.FormatV2, &myHashOptions)
-	if err1 == nil {
-		pol2Hash, err2 := hashstructure.Hash(policy2, hashstructure.FormatV2, &myHashOptions)
-		if err2 == nil {
-			return pol1Hash == pol2Hash
-		}
+// isSame compares given hash string with the hash of given interface and return true
+// if these values are same; otherwise, returns false.
+func isSame(latestHash string, i interface{}) bool {
```

Comment:
If we break the "calculateHashes" function in a single parameter-single variable as suggested, this function can be removed all together and you cna use string == calculateHash(x)

Reply:
Thanks! Implemented in [0365448](https://github.com/TykTechnologies/tyk-operator/pull/625/commits/0365448d7f6ea2d6ea9cf16a4aa4bc1a16a72ec7)